### ISSUE TITLE
The replay buffer is not reset

### ISSUE THREAD

**Issue by Tamago4a** [obsproject/obs-studio#9132]
### Operating System Info

Windows 10

### OBS Studio Version

29.1.3

### OBS Studio Log URL

https://obsproject.com/logs/cyE5zaWrZGfFHufK

### Expected Behavior

I expected that I would not see those video moments that were already in the saved replay in the new replays.

### Current Behavior

However, this did not happen. Each new saved replay included everything from the moment the replay was enabled. When the maximum replay time approached the maximum, only then the video moments from the beginning of the replay began to be erased and did not fall into new replays.

NVIDIA Shadowplay has the same feature, but parts of replays that are already saved are not saved in new replays.

### Steps to Reproduce

1. Just save two reps 10 seconds apart and you will see that both reps will have the same start.

**Comment by Fenrirthviti** [comment]
This is currently by design.

You can show your support for discard buffer on save here: https://ideas.obsproject.com/posts/1022/discard-replay-buffer-on-save

**Comment by Tamago4a** [comment]
> This is currently by design.
> 
> You can show your support for discard buffer on save here: https://ideas.obsproject.com/posts/1022/discard-replay-buffer-on-save

I expected it to be on purpose. This doesn't look like a bug. I voted in the link you provided. In fact, I like how it works at the moment. Sometimes I save a replay and then realize that I saved it too soon. And if this is done in nvidia shadow play, then between repetitions there will be a micro-break that interferes with normal video editing and further use. But saving for all 2 and a half minutes (as I have configured) is stupid. Well, they would leave 5 seconds from the old repeat, and that would be enough. In any case, this setting should be configured more flexibly, and not as it is now.
